ellama.el

```diff
@@ -100,6 +100,76 @@
   (rx (minimal-match
        (literal "```") (zero-or-more anything))))
 
+(defun ellama-set-visual-line-mode-for-ellama-buffer ()
+  "Set `visual-line-mode' for the *ellama* buffer."
+  (when (string-prefix-p "*ellama*" (buffer-name))
+    (visual-line-mode 1)))
+
+(defcustom ellama-enable-break-lines t
+  "Enable or disable break lines for *ellama* buffer."
+  :type 'boolean
+  :group 'ellama
+  :set (lambda (symbol value)
+         (set symbol value)
+         (if value
+             (add-hook 'buffer-list-update-hook 'ellama-set-visual-line-mode-for-ellama-buffer)
+           (remove-hook 'buffer-list-update-hook 'ellama-set-visual-line-mode-for-ellama-buffer))))
+
+(defun ellama-setup-keymap ()
+  "Set up the Ellama keymap and bindings."
+  (interactive)
+  (defvar ellama-keymap (make-sparse-keymap)
+    "Keymap for Ellama Commands")
+
+  (define-key global-map (kbd ellama-keymap-prefix) ellama-keymap)
+
+  (let ((key-commands
+          '(
+             ;; code
+	           ("c c" ellama-code-complete "Code complete")
+	           ("c a" ellama-code-add "Code add")
+	           ("c e" ellama-code-edit "Code edit")
+	           ("c i" ellama-code-improve "Code improve")
+	           ("c r" ellama-code-review "Code review")
+	           ;; summarize
+	           ("s s" ellama-summarize "Summarize")
+	           ("s w" ellama-summarize-webpage "Summarize webpage")
+	           ;; improve
+	           ("i w" ellama-improve-wording "Improve wording")
+	           ("i g" ellama-improve-grammar "Improve grammar and spelling")
+	           ("i c" ellama-improve-conciseness "Improve conciseness")
+	           ;; make
+	           ("m l" ellama-make-list "Make list")
+	           ("m t" ellama-make-table "Make table")
+	           ("m f" ellama-make-format "Make format")
+	           ;; ask
+	           ("a a" ellama-ask-about "Ask about")
+	           ("a i" ellama-ask-interactive "Chat with ellama")
```

Comment:
for example, instead of "Chat with ellama" use "Ask interactively" or something like that

Reply:
Noice!